bin/src/extract.rs

```diff
@@ -174,6 +191,67 @@ pub async fn main(opts: Options) -> Result<()> {
         .await
         .context("configure remote registry")?;
 
+    extract_layers(opts, registry)
+        .await
+        .context("extract layers")
+}
+
+async fn strategy_daemon(opts: &Options) -> Result<()> {
+    if opts.target.is_path() {
+        debug!("input appears to be a file path, skipping strategy");
+        return Ok(());
+    }
+
+    let layer_globs = Filters::parse_glob(opts.layer_glob.iter().flatten())?;
+    let file_globs = Filters::parse_glob(opts.file_glob.iter().flatten())?;
+    let layer_regexes = Filters::parse_regex(opts.layer_regex.iter().flatten())?;
+    let file_regexes = Filters::parse_regex(opts.file_regex.iter().flatten())?;
+
+    let daemon = Daemon::builder()
+        .reference(&opts.target.image)
+        .layer_filters(layer_globs + layer_regexes)
+        .file_filters(file_globs + file_regexes)
+        .build()
+        .await
+        .context("build daemon reference")?;
+
+    tracing::info!("pulled image from daemon");
+    extract_layers(opts, daemon).await.context("extract layers")
+}
+
+async fn strategy_tarball(opts: &Options) -> Result<()> {
+    let path = PathBuf::from(&opts.target.image);
+    if matches!(tokio::fs::try_exists(&path).await, Err(_) | Ok(false)) {
+        bail!("path does not exist: {path:?}");
+    }
+
+    let layer_globs = Filters::parse_glob(opts.layer_glob.iter().flatten())?;
+    let file_globs = Filters::parse_glob(opts.file_glob.iter().flatten())?;
+    let layer_regexes = Filters::parse_regex(opts.layer_regex.iter().flatten())?;
+    let file_regexes = Filters::parse_regex(opts.file_regex.iter().flatten())?;
```

Comment:
[Optional] If the layer handling is truly the same between strategies and is likely to be in the future I think it'd be worth putting these into a helper function.